pandas/tests/indexes/test_base.py

```diff
@@ -787,21 +796,63 @@ def test_chained_union(self, sort):
         expected = j1.union(j2, sort=sort).union(j3, sort=sort)
         tm.assert_index_equal(union, expected)
 
-    @pytest.mark.parametrize("sort", [True, False])
+    @pytest.mark.parametrize("sort", [None, False])
     def test_union(self, sort):
         # TODO: Replace with fixturesult
         first = self.strIndex[5:20]
         second = self.strIndex[:10]
         everything = self.strIndex[:20]
 
         union = first.union(second, sort=sort)
-        if sort:
+        if sort is None:
             tm.assert_index_equal(union, everything.sort_values())
         assert tm.equalContents(union, everything)
 
+    @pytest.mark.parametrize('slice_', [slice(None), slice(0)])
+    def test_union_sort_other_special(self, slice_):
+        # https://github.com/pandas-dev/pandas/issues/24959
+
+        idx = pd.Index([1, 0, 2])
+        # default, sort=None
+        other = idx[slice_]
+        tm.assert_index_equal(idx.union(other), idx)
+        tm.assert_index_equal(other.union(idx), idx)
+
+        # sort=False
+        tm.assert_index_equal(idx.union(other, sort=False), idx)
+
+        # TODO decide on True behaviour
```

Comment:
why are commenting these? is True not a legitimate value? and doesn't it actually work like this now? This is very strange.

Reply:
I thought that’s what you wanted in the other PR. It’s what you agreed to in https://github.com/pandas-dev/pandas/pull/25007#issuecomment-459385344

Reply:
right, but does ``sort=True`` do *anything* now (is it accepted)?

Reply:
It raises. This PR is just changing the keyword for the currently implemented behavior from True to None. It’s not adding anything that wasn’t previously implemented (sorting always).

Reply:
It is explicitly disallowed (meaning: it will raise an error if passed), only None and False are accepted

Reply:
oh ok, that's fine. are you explicity testing that? (I realize why you are having the commented out ones, *maybe* better to xfail those though)

Reply:
Moved the `true` cases to their own tests and failed them in 2e181ac